src/monmove.cpp

```diff
@@ -1442,3 +1572,31 @@ int monster::turns_to_reach( int x, int y )
 
     return int( turns + .9 ); // Halve (to get turns) and round up
 }
+
+std::pair<item *, tripoint> monster::get_most_desired_item_in_radius( int radius,
+        std::function<float( const item & )> calc_desirability ) const
+{
+    float max_desirability = -1;
+    tripoint desired_item_pos;
+    item *desired_item = nullptr;
+    for( auto &p : g->m.points_in_radius( pos(), radius ) ) {
+        if( g->m.sees_some_items( p, *this ) ) {
+            auto items = g->m.i_at( p );
+            for( auto &itm : items ) {
+                float des = calc_desirability( itm );
+                if( des > max_desirability ) {
+                    desired_item = &itm;
+                    desired_item_pos = p;
+                }
+            }
+        }
+    }
+    return std::make_pair( desired_item, desired_item_pos );
+}
+
+std::pair<item *, tripoint> monster::get_most_desired_visible_item(
+    std::function<float( const item & )> calc_desirability ) const
+{
+    int sight_radius = sight_range( g->m.ambient_light_at( pos() ) );
+    return get_most_desired_item_in_radius( sight_radius, calc_desirability );
```

Comment:
Can `get_most_desired_item_in_radius()` be used separately? If no, I'd merge these functions into one.

Reply:
Yes, I intended to keep it separate in case future implementors wanted to search in a radius other than the sight radius.